qcsubmit/workflow_components/state_enumeration.py

```diff
@@ -169,30 +204,27 @@ def apply(self, molecules: List[Molecule]) -> ComponentResult:
             This is only possible using Openeye so far, if openeye is not available this step will fail.
         """
 
-        from openforcefield.utils.toolkits import OpenEyeToolkitWrapper
+        result = self._create_result(skip_unique_check=self.skip_unique_check)
 
-        result = self._create_result()
+        has_oe = self.cache
 
         # must have openeye to use this feature
-        if OpenEyeToolkitWrapper.is_available():
+        if has_oe:
 
             for molecule in molecules:
-
                 try:
                     protomers = molecule.enumerate_protomers(max_states=self.max_states)
 
-                    for protomer in protomers:
-                        result.add_molecule(protomer)
-                    result.add_molecule(molecule)
-
-                except Exception:
-                    self.fail_molecule(molecule=molecule, component_result=result)
+                    if len(protomers) == 0:
+                        result.add_molecule(molecule)
 
-            return result
+                    else:
+                        for protomer in protomers:
```

Comment:
we need to be careful here as we may lose the input protomer as the toolkit removes them from the return.

Reply:
Submitted https://github.com/openforcefield/openforcefield/issues/738 to see if we can get some blanket clarification in the TK

Reply:
as long as it hasn't been changed since I put the state enumeration methods in the toolkit they all should remove the input molecule regardless of the backend toolkit used see here for the molecule class docstrings https://github.com/openforcefield/openforcefield/blob/47d98f49109338ec1c29b907f98ace4d1fdbbb53/openforcefield/topology/molecule.py#L3947.

Reply:
Ah! I was careless and only read the main docstring. Didn't think I had to read the return data description. 
That said, though, I do think we are getting the "same" input back. Going to carry the conversation on to #738